cmd/server-mux.go

```diff
@@ -420,21 +443,97 @@ func (m *ServerMux) ListenAndServe(certFile, keyFile string) (err error) {
 		MinVersion: tls.VersionTLS12,
 	} // Always instantiate.
 
-	if tlsEnabled {
-		// Configure TLS in the server
-		if config.NextProtos == nil {
-			config.NextProtos = []string{"http/1.1", "h2"}
+	// Configure TLS in the server
+	if config.NextProtos == nil {
+		config.NextProtos = []string{"http/1.1", "h2"}
+	}
+	config.Certificates = make([]tls.Certificate, 1)
+
+	config.Certificates[0], tlsErr = tls.X509KeyPair(m.certPEM, m.keyPEM)
+	if tlsErr != nil {
+		return nil, false, false, tlsErr
+	}
+
+	// Fetch for OCSP response
+	ocspBytes, _, ocspErr := GetOCSPForCert(m.certPEM)
+	if ocspErr == nil {
+		ocspUpdated = true
+		config.Certificates[0].OCSPStaple = ocspBytes
+	} else {
+		if strings.Contains(ocspErr.Error(), "no OCSP server specified in cert") {
+			ocspUnsupported = true
 		}
-		config.Certificates = make([]tls.Certificate, 1)
-		config.Certificates[0], err = tls.LoadX509KeyPair(certFile, keyFile)
-		if err != nil {
-			return err
+	}
+
+	// Save the build tls config
+	m.tlsMu.Lock()
+	m.tlsConfig = config
+	m.tlsMu.Unlock()
+
+	return config, ocspUpdated, ocspUnsupported, nil
+}
+
+// tlsBackgroundUpdate - regularly builds tls config to update
+// ocsp response.
+func (m *ServerMux) tlsBackgroundUpdate() {
```

Comment:
> Why is this functionality as method to ServerMux?

ServerMux provides tls certificates to clients when tls is activated. This method updates the tls certificate so it will be tied to the ServerMux instance.